packages/styled-components/src/models/StyledComponent.js

```diff
@@ -206,6 +209,17 @@ export default function createStyledComponent(
       ? Array.prototype.concat(target.attrs, attrs).filter(Boolean)
       : attrs;
 
+  let { shouldForwardProp } = options;
+  // $FlowFixMe
+  if (isTargetStyledComp && target.shouldForwardProp) {
+    if (shouldForwardProp) {
+      // compose nested shouldForwardProp calls
+      // $FlowFixMe
+      shouldForwardProp = prop => target.shouldForwardProp(prop) && options.shouldForwardProp(prop);
+    }
+    ({ shouldForwardProp } = target);
```

Comment:
This doesn’t seem right maybe, 🤔 was this supposed to be in an `else` statement?

nit: Also can we drop the destructuring here please? We’re also not destructuring for the arrow function composition above, so let’s maybe settle on one pattern for both?

Reply:
heh - i didn't want to use destructuring here - eslint complained at me :wink: happy to drop

Reply:
and yes - should have had an `else` in there - good spot :grin: